src/Datadog.Trace/DuckTyping/DuckType.Statics.cs

```diff
@@ -58,24 +58,26 @@ private static ModuleBuilder GetModuleBuilder(Type targetType)
                 {
                     if (type.Assembly != targetAssembly)
                     {
-                        var name = new AssemblyName($"DuckTypeGenericTypeAssembly.{targetType.Name}_{++_assemblyCount}");
-                        name.Version = targetAssembly.GetName().Version;
-                        var assemblyBuilder = AssemblyBuilder.DefineDynamicAssembly(name, AssemblyBuilderAccess.Run);
-                        return assemblyBuilder.DefineDynamicModule("MainModule");
+                        return CreateModuleBuilder($"DuckTypeGenericTypeAssembly.{targetType.Name}_{++_assemblyCount}", targetAssembly);
```

Comment:
I would have put the `_assemblyCount` suffix inside of the helper method, but oh well 😄

Reply:
done in 8bb619b